src/tech/v3/ml.clj

```diff
@@ -49,6 +52,12 @@
   (:hyperparameters (options->model-def {:model-type model-kwd})))
 
 
+(defn preprocess [dataset options]
+  (let [fun (get-in options [:preprocess :fn] identity)]
```

Comment:
Maybe the ":preprocess" submap does not make sense here, and the pre-processing related keys should be on top level of the options map ?

Reply:
I would like them to be at the top level.  There aren't that many keys in the map as is.  I expect most people will want to serialize options with the model, so whatever is in the options map will need to be, in some cases serializable.

I think if :preprocess-fn can also be a namespaced symbol or keyword we can do 'requiring-resolve' to resolve to the actual preprocess-fn.  This also allows it to be used in a distributive fashion.

Reply:
I did not think about distributed execution.
Is there some infrastructure or example for this somewhere ?

Reply:
I have implemented it, as you suggested